### PR TITLE
Get rid of PAGE_SIZE usage in user-space code

### PR DESCRIPTION
**Pull request by andriytk**
Not all platforms have PAGE_SIZE defined in user-space.
(For example, CentOS-7 aarch64 does not define it.)

Solution: 1) use m0_pagesize_get() for PAGE_SIZE and,
similarly, introduce function m0_pageshift_get() and
use it for PAGE_SHIFT. 2) Avoid defining structures
and buffers based in PAGE_SIZE.

### CODE REVIEW

**Review by huanghua78 — APPROVED** [review]

File: lib/trace.c [hunk 1]
```diff
@@ -70,13 +69,15 @@ static const char     trace_magic_symbol_name[] = "trace_magic_symbol";
 
 /* single buffer for now */
 
+enum {BOOTLOG_BUF_SIZE = (1 << 16)}; /* 64K */
```
Comment (ivan-alekhin) [inline lib/trace.c:72]: \*nitpicking\*  Spaces after `{` and before `}`. See M0_LOCALITY_LOCKERS_NR.

\*suggestion\* `/* The bootlog buffer size should be aligned with the page size. See the preconditions in m0_trace_init. */` I'm not sure though if it is really necessary.

**Review by ivan-alekhin — APPROVED** [review]
+1 and a minor suggestion.

What I did: reviewed the code around the places that were changed, fetched the patch, checked formatting in the patch, and built the code with the patch applied.
What I did not do: did not run the tests, did not dig into the big picture (performance impact and so on).

File: lib/trace.c (hunk 1 above)
Comment (andriytk) [reply lib/trace.c:72]: Agree, added. Thanks!

**Comment by andriytk** [comment]
Started http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810/.

**Comment by cortx-admin** [comment]
<h1>Jenkins CI Result : <a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810/>Motr#810</a></h1><h2>Motr Test Summary</h2><table><tr><th>Test Result</th><th>Count</th><th>Info</tr><tr><td>:x:Failed</td><td>2</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/01motr-single-node/52motr-singlenode-sanity target=_blank>01motr-single-node/52motr-singlenode-sanity</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/01motr-single-node/00userspace-tests target=_blank>01motr-single-node/00userspace-tests</a></p></details></td></tr><tr><td>:checkered_flag:Skipped</td><td>5</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/01motr-single-node/37protocol target=_blank>01motr-single-node/37protocol</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/04motr-single-node/28sys-kvs-kernel target=_blank>04motr-single-node/28sys-kvs-kernel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/05motr-single-node/07mount-fail target=_blank>05motr-single-node/07mount-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/05motr-single-node/07mount-multiple target=_blank>05motr-single-node/07mount-multiple</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/05motr-single-node/17sns-repair-concurrent-io target=_blank>05motr-single-node/17sns-repair-concurrent-io</a></p></details></td></tr><tr><td>:heavy_check_mark:Passed</td><td>57</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/01motr-single-node/01kernel-tests target=_blank>01motr-single-node/01kernel-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/01motr-single-node/01net target=_blank>01motr-single-node/01net</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/01motr-single-node/02rpcping target=_blank>01motr-single-node/02rpcping</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/01motr-single-node/03console target=_blank>01motr-single-node/03console</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/01motr-single-node/04initscripts target=_blank>01motr-single-node/04initscripts</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/01motr-single-node/05confgen target=_blank>01motr-single-node/05confgen</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/01motr-single-node/06hagen target=_blank>01motr-single-node/06hagen</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/01motr-single-node/28sys-kvs target=_blank>01motr-single-node/28sys-kvs</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/01motr-single-node/35m0singlenode target=_blank>01motr-single-node/35m0singlenode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/01motr-single-node/43m0crate target=_blank>01motr-single-node/43m0crate</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/02motr-single-node/07m0d-fatal target=_blank>02motr-single-node/07m0d-fatal</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/02motr-single-node/08spiel target=_blank>02motr-single-node/08spiel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/02motr-single-node/10pver-assign target=_blank>02motr-single-node/10pver-assign</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/02motr-single-node/11m0t1fs target=_blank>02motr-single-node/11m0t1fs</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/02motr-single-node/13dgmode-io target=_blank>02motr-single-node/13dgmode-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/02motr-single-node/14poolmach target=_blank>02motr-single-node/14poolmach</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/02motr-single-node/20rpc-session-cancel target=_blank>02motr-single-node/20rpc-session-cancel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/02motr-single-node/21fsync-single-node target=_blank>02motr-single-node/21fsync-single-node</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/02motr-single-node/26motr-user-kernel-tests target=_blank>02motr-single-node/26motr-user-kernel-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/02motr-single-node/41motr-conf-update target=_blank>02motr-single-node/41motr-conf-update</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/02motr-single-node/51kem target=_blank>02motr-single-node/51kem</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/02motr-single-node/53clusterusage-alert target=_blank>02motr-single-node/53clusterusage-alert</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/03motr-single-node/04sss target=_blank>03motr-single-node/04sss</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/03motr-single-node/06conf target=_blank>03motr-single-node/06conf</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/03motr-single-node/08spiel-multi-confd target=_blank>03motr-single-node/08spiel-multi-confd</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/03motr-single-node/24motr-dix-repair-lookup-insert target=_blank>03motr-single-node/24motr-dix-repair-lookup-insert</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/03motr-single-node/36spare-reservation target=_blank>03motr-single-node/36spare-reservation</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/03motr-single-node/60sns-repair-motr-1f target=_blank>03motr-single-node/60sns-repair-motr-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/03motr-single-node/61sns-repair-motr-1n-1f target=_blank>03motr-single-node/61sns-repair-motr-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/03motr-single-node/62sns-repair-motr-mf target=_blank>03motr-single-node/62sns-repair-motr-mf</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/03motr-single-node/63sns-repair-motr-1k-1f target=_blank>03motr-single-node/63sns-repair-motr-1k-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/03motr-single-node/64sns-repair-motr-ios-fail target=_blank>03motr-single-node/64sns-repair-motr-ios-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/03motr-single-node/65sns-repair-motr-abort target=_blank>03motr-single-node/65sns-repair-motr-abort</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/03motr-single-node/66sns-repair-motr-abort-quiesce target=_blank>03motr-single-node/66sns-repair-motr-abort-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/04motr-single-node/08spiel-sns-repair target=_blank>04motr-single-node/08spiel-sns-repair</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/04motr-single-node/08spiel-sns-repair-quiesce target=_blank>04motr-single-node/08spiel-sns-repair-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/04motr-single-node/11m0t1fs-rconfc-fail target=_blank>04motr-single-node/11m0t1fs-rconfc-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/04motr-single-node/19sns-repair-abort target=_blank>04motr-single-node/19sns-repair-abort</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/04motr-single-node/22sns-repair-ios-fail target=_blank>04motr-single-node/22sns-repair-ios-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/04motr-single-node/25m0kv target=_blank>04motr-single-node/25m0kv</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/04motr-single-node/34sns-repair-1n-1f target=_blank>04motr-single-node/34sns-repair-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/04motr-single-node/44motr-rm-lock-cc-io target=_blank>04motr-single-node/44motr-rm-lock-cc-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/04motr-single-node/45motr-rmw target=_blank>04motr-single-node/45motr-rmw</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/04motr-single-node/48motr-raid0-io target=_blank>04motr-single-node/48motr-raid0-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/04motr-single-node/49motr-rpc-cancel target=_blank>04motr-single-node/49motr-rpc-cancel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/05motr-single-node/07mount target=_blank>05motr-single-node/07mount</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/05motr-single-node/12fsync target=_blank>05motr-single-node/12fsync</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/05motr-single-node/12fwait target=_blank>05motr-single-node/12fwait</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/05motr-single-node/15sns-repair-single target=_blank>05motr-single-node/15sns-repair-single</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/05motr-single-node/16sns-repair-multi target=_blank>05motr-single-node/16sns-repair-multi</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/05motr-single-node/18sns-repair-quiesce target=_blank>05motr-single-node/18sns-repair-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/05motr-single-node/23sns-abort-quiesce target=_blank>05motr-single-node/23sns-abort-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/05motr-single-node/40motr-dgmode target=_blank>05motr-single-node/40motr-dgmode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/05motr-single-node/42motr-utils target=_blank>05motr-single-node/42motr-utils</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/05motr-single-node/43motr-sync-replication target=_blank>05motr-single-node/43motr-sync-replication</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/05motr-single-node/44motr-sns-repair target=_blank>05motr-single-node/44motr-sns-repair</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit/05motr-single-node/45motr-sns-repair-N-1 target=_blank>05motr-single-node/45motr-sns-repair-N-1</a></p></details></td></tr><tr><td>Total</td><td>64</td><td><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/810//testReport/junit>:link:</a></td></tr></table><h2> CppCheck Summary</h2><p> &emsp;&emsp;&emsp;Cppcheck: No new warnings found :thumbsup:</p>

**Comment by andriytk** [comment]
Unfortunately, it fails on CentOS 7 aarch64:

```
$ scripts/m0 run-ut -t libm0-ut
----- run_ut -t libm0-ut -----
motr[01703]:  a3d0  FATAL  [lib/assert.c:50:m0_panic]  panic: ((trace_area_size % m0_pagesize_get()) == 0) at logbuf_map() (lib/user_space/utrace.c:68)  [git: 2.0.0-307-109-gb6a5dae] 
Motr panic: ((trace_area_size % m0_pagesize_get()) == 0) at logbuf_map() lib/user_space/utrace.c:68 (errno: 2) (last failed: none) [git: 2.0.0-307-109-gb6a5dae] pid: 1703  
```
@chumakd, do you remember - why do we need that precondition there?

**Comment by chumakd** [comment]
> Unfortunately, it fails on CentOS 7 aarch64:
> 
> ```
> $ scripts/m0 run-ut -t libm0-ut
> ----- run_ut -t libm0-ut -----
> motr[01703]:  a3d0  FATAL  [lib/assert.c:50:m0_panic]  panic: ((trace_area_size % m0_pagesize_get()) == 0) at logbuf_map() (lib/user_space/utrace.c:68)  [git: 2.0.0-307-109-gb6a5dae] 
> Motr panic: ((trace_area_size % m0_pagesize_get()) == 0) at logbuf_map() lib/user_space/utrace.c:68 (errno: 2) (last failed: none) [git: 2.0.0-307-109-gb6a5dae] pid: 1703  
> ```
> 
> @chumakd, do you remember - why do we need that precondition there?

I think that was added to ensure that full content of trace area to be written to the trace file. The trace file is memory mapped with `mmap(2)` and according to the manpage, if size of the mapped region is not multiple of `PAGE_SIZE`, some of the data changed in the mapped memory may not get into the file:

> A file is mapped in multiples of the page size.  For a file that is not a multiple of the page size, the remaining memory is zeroed when mapped, and writes to that region are  not written out to the file.

Thus, if the precondition doesn't hold, some of the trace records at the end of the buffer may be missing in the trace file on disk. But that needs to be tested.

**Review by madhavemuri — APPROVED** [review]

**Comment by andriytk** [comment]
@chumakd, how about stealing some part of the buffer for the header accommodation and making sure that the resulting header + buffer size is multiple of page size? See my last commit here.

**Comment by andriytk** [comment]
Started http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/818/

**Comment by andriytk** [comment]
Started http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819/.

**Comment by cortx-admin** [comment]
<h1>Jenkins CI Result : <a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819/>Motr#819</a></h1><h2>Motr Test Summary</h2><table><tr><th>Test Result</th><th>Count</th><th>Info</tr><tr><td>:x:Failed</td><td>2</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/01motr-single-node/52motr-singlenode-sanity target=_blank>01motr-single-node/52motr-singlenode-sanity</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/02motr-single-node/41motr-conf-update target=_blank>02motr-single-node/41motr-conf-update</a></p></details></td></tr><tr><td>:checkered_flag:Skipped</td><td>5</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/01motr-single-node/37protocol target=_blank>01motr-single-node/37protocol</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/04motr-single-node/28sys-kvs-kernel target=_blank>04motr-single-node/28sys-kvs-kernel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/05motr-single-node/07mount-fail target=_blank>05motr-single-node/07mount-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/05motr-single-node/07mount-multiple target=_blank>05motr-single-node/07mount-multiple</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/05motr-single-node/17sns-repair-concurrent-io target=_blank>05motr-single-node/17sns-repair-concurrent-io</a></p></details></td></tr><tr><td>:heavy_check_mark:Passed</td><td>57</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/01motr-single-node/00userspace-tests target=_blank>01motr-single-node/00userspace-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/01motr-single-node/01kernel-tests target=_blank>01motr-single-node/01kernel-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/01motr-single-node/01net target=_blank>01motr-single-node/01net</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/01motr-single-node/02rpcping target=_blank>01motr-single-node/02rpcping</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/01motr-single-node/03console target=_blank>01motr-single-node/03console</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/01motr-single-node/04initscripts target=_blank>01motr-single-node/04initscripts</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/01motr-single-node/05confgen target=_blank>01motr-single-node/05confgen</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/01motr-single-node/06hagen target=_blank>01motr-single-node/06hagen</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/01motr-single-node/28sys-kvs target=_blank>01motr-single-node/28sys-kvs</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/01motr-single-node/35m0singlenode target=_blank>01motr-single-node/35m0singlenode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/01motr-single-node/43m0crate target=_blank>01motr-single-node/43m0crate</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/02motr-single-node/07m0d-fatal target=_blank>02motr-single-node/07m0d-fatal</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/02motr-single-node/08spiel target=_blank>02motr-single-node/08spiel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/02motr-single-node/10pver-assign target=_blank>02motr-single-node/10pver-assign</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/02motr-single-node/11m0t1fs target=_blank>02motr-single-node/11m0t1fs</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/02motr-single-node/13dgmode-io target=_blank>02motr-single-node/13dgmode-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/02motr-single-node/14poolmach target=_blank>02motr-single-node/14poolmach</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/02motr-single-node/20rpc-session-cancel target=_blank>02motr-single-node/20rpc-session-cancel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/02motr-single-node/21fsync-single-node target=_blank>02motr-single-node/21fsync-single-node</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/02motr-single-node/26motr-user-kernel-tests target=_blank>02motr-single-node/26motr-user-kernel-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/02motr-single-node/51kem target=_blank>02motr-single-node/51kem</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/02motr-single-node/53clusterusage-alert target=_blank>02motr-single-node/53clusterusage-alert</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/03motr-single-node/04sss target=_blank>03motr-single-node/04sss</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/03motr-single-node/06conf target=_blank>03motr-single-node/06conf</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/03motr-single-node/08spiel-multi-confd target=_blank>03motr-single-node/08spiel-multi-confd</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/03motr-single-node/24motr-dix-repair-lookup-insert target=_blank>03motr-single-node/24motr-dix-repair-lookup-insert</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/03motr-single-node/36spare-reservation target=_blank>03motr-single-node/36spare-reservation</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/03motr-single-node/60sns-repair-motr-1f target=_blank>03motr-single-node/60sns-repair-motr-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/03motr-single-node/61sns-repair-motr-1n-1f target=_blank>03motr-single-node/61sns-repair-motr-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/03motr-single-node/62sns-repair-motr-mf target=_blank>03motr-single-node/62sns-repair-motr-mf</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/03motr-single-node/63sns-repair-motr-1k-1f target=_blank>03motr-single-node/63sns-repair-motr-1k-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/03motr-single-node/64sns-repair-motr-ios-fail target=_blank>03motr-single-node/64sns-repair-motr-ios-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/03motr-single-node/65sns-repair-motr-abort target=_blank>03motr-single-node/65sns-repair-motr-abort</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/03motr-single-node/66sns-repair-motr-abort-quiesce target=_blank>03motr-single-node/66sns-repair-motr-abort-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/04motr-single-node/08spiel-sns-repair target=_blank>04motr-single-node/08spiel-sns-repair</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/04motr-single-node/08spiel-sns-repair-quiesce target=_blank>04motr-single-node/08spiel-sns-repair-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/04motr-single-node/11m0t1fs-rconfc-fail target=_blank>04motr-single-node/11m0t1fs-rconfc-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/04motr-single-node/19sns-repair-abort target=_blank>04motr-single-node/19sns-repair-abort</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/04motr-single-node/22sns-repair-ios-fail target=_blank>04motr-single-node/22sns-repair-ios-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/04motr-single-node/25m0kv target=_blank>04motr-single-node/25m0kv</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/04motr-single-node/34sns-repair-1n-1f target=_blank>04motr-single-node/34sns-repair-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/04motr-single-node/44motr-rm-lock-cc-io target=_blank>04motr-single-node/44motr-rm-lock-cc-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/04motr-single-node/45motr-rmw target=_blank>04motr-single-node/45motr-rmw</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/04motr-single-node/48motr-raid0-io target=_blank>04motr-single-node/48motr-raid0-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/04motr-single-node/49motr-rpc-cancel target=_blank>04motr-single-node/49motr-rpc-cancel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/05motr-single-node/07mount target=_blank>05motr-single-node/07mount</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/05motr-single-node/12fsync target=_blank>05motr-single-node/12fsync</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/05motr-single-node/12fwait target=_blank>05motr-single-node/12fwait</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/05motr-single-node/15sns-repair-single target=_blank>05motr-single-node/15sns-repair-single</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/05motr-single-node/16sns-repair-multi target=_blank>05motr-single-node/16sns-repair-multi</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/05motr-single-node/18sns-repair-quiesce target=_blank>05motr-single-node/18sns-repair-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/05motr-single-node/23sns-abort-quiesce target=_blank>05motr-single-node/23sns-abort-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/05motr-single-node/40motr-dgmode target=_blank>05motr-single-node/40motr-dgmode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/05motr-single-node/42motr-utils target=_blank>05motr-single-node/42motr-utils</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/05motr-single-node/43motr-sync-replication target=_blank>05motr-single-node/43motr-sync-replication</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/05motr-single-node/44motr-sns-repair target=_blank>05motr-single-node/44motr-sns-repair</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit/05motr-single-node/45motr-sns-repair-N-1 target=_blank>05motr-single-node/45motr-sns-repair-N-1</a></p></details></td></tr><tr><td>Total</td><td>64</td><td><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/819//testReport/junit>:link:</a></td></tr></table><h2> CppCheck Summary</h2><p> &emsp;&emsp;&emsp;Cppcheck: No new warnings found :thumbsup:</p>

**Comment by chumakd** [comment]
> @chumakd, how about 1) stealing some part of the buffer for the header accommodation and 2) making sure that the resulting header + buffer size is multiple of page size? See my last commit here.

@andriytk I think that can work, though it complicates the implementation IMO. In any case I would advise to at a unit test into `m0-ut` that will check for possible corner cases with log buffer overflow and make sure no trace records are missing in the underlying mmaped file.

**Comment by andriytk** [comment]
I decided to make it simpler and less risky way - set the header size multiple of page size on all currently supported platforms (which makes it 64KB).

**Comment by andriytk** [comment]
@mehjoshi, please merge it without squashing. Thanks.
